File: ClientTester.java
import java.util.Scanner;

public class ClientTester {

    static ClientSidePlayer player;

    public static void main(String[] args) throws Exception {
        InputParser p = new InputParser();
        p.start();
        player = new ClientSidePlayer("192.168.0.24");
        //p.stahp();
    }

    static class InputParser extends Thread {

        Scanner input;
        boolean stopped;
        static int num = 1;

        public InputParser() {
            super("ClientTester$InputParser-" + num);
            num++;
            input = new Scanner(System.in);
            stopped = false;
        }

        @Override
        public void run() {
            while (!stopped) {
                if (input.hasNext()) {
                    player.client().transmit(input.nextLine());
                }
            }
        }

        public void stahp() {
            stopped = true;
        }
    }
}

File: src/main/java/com/isaac/games/ClientTester.kt
package com.isaac.games

import java.util.Scanner

object ClientTester {

    lateinit var player: ClientSidePlayer

    @Throws(Exception::class)
    @JvmStatic
    fun main(args: Array<String>) {
        val p = InputParser()
        p.start()
        player = ClientSidePlayer("127.0.0.1")
        //p.stahp();
    }

    internal class InputParser : Thread("ClientTester\$InputParser-$num") {

        var input: Scanner
        var stopped: Boolean = false

        init {
            num++
            input = Scanner(System.`in`)
            stopped = false
        }

        override fun run() {
            while (!stopped) {
                if (input.hasNext()) {
                    player.client().transmit(input.nextLine())
                }
            }
        }

        fun stahp() {
            stopped = true
        }

        companion object {
            var num = 1
        }
    }
}